src/Fields/mapreduce.jl

```diff
@@ -67,24 +67,32 @@ end
 """
     norm(v::Field, p=2; normalize=true)
 
-The approximate ``L^p`` norm of `v`. This is computed by summation of the norms
-of the field values multiplied by the Jacobians and the quadrature weights. If
-`normalize=true` (the default), then it is divided by the sum of the weights:
+The approximate ``L^p`` norm of `v`, where ``L^p`` represents the space of measurable
+functions for which the p-th power of the absolute value is Lebesgue integrable, that is:
 ```math
-\\left(\\frac{\\sum_i \\|v_i\\|^p W_i J_i}{\\sum_i W_i J_i}\\right)^{1/p}
+\\| v \\|_p = \\left( \\int_\\Omega |v_i|^p d \\Omega \\right)^{1/p}
+```
+where ``v_i`` is the value of each component of the `Field` ``v``, in case of vector-valued fields,
```

Comment:
`i` below refers to the nodes at which we have values, not components.

Reply:
That's why I said if we wanted to be more precise we should use two indices, one for the components and one for the node values. How would you want to describe this in a way that is mathematically correct to describe the L^p function norm and reflects the code?

Reply:
the way it is written in the `LinearAlgebra.norm` docs

Reply:
I think it might make more sense to draw the distinction between the field as a function on Omega, and as a discrete set of node values. I don't think you need to talk about components, other than to say that `|v(x)|` is defined to be the absolute value if scalar-values, or the 2-norm if a vector or composite valued.

Reply:
The way they write it is suitable for array-like objects (l^2 discrete [sequences](https://en.wikipedia.org/wiki/Lp_space#General_%E2%84%93p-space)) but it is not technically suitable for [L^2 function spaces](https://en.wikipedia.org/wiki/Lp_space#Lp_spaces_and_Lebesgue_integrals), that we are defining here in a function space. I know we discretely represent the function as a Field, but since we use the integral in the definition (unlike the `LinearAlgebra.norm` docs) we have to be careful and consistent. 
![image](https://user-images.githubusercontent.com/39932030/152420912-6ef155cd-66a5-4212-a505-5cce333f6da1.png)
![image](https://user-images.githubusercontent.com/39932030/152421366-a833eaff-0373-4df2-8138-e3a2a0d2564f.png)

Reply:
@simonbyrne I reworded this following your suggestion and squashed. I also agree that we should definitely draw the distinction between the field as a function on Omega, and as a discrete set of node values. But it should perhaps be done in a broader introduction to the (Fields?) API, not in the norm function docstring